### ISSUE TITLE
Upgrade broken from 1.6.8 to 1.6.16

### ISSUE THREAD

**Issue by migratis** [jelix/jelix#322]
Hello,

when upgrading from Jelix 1.6.8 to jelix 1.6.16 the command line  :

#php installer.php 

stops with this errors   :

Installation start..
[notice] Installation starts for the entry point index
[error] bad version number : comp=jacl2
[error] bad version number : comp=jacl2db
[error] bad version number : comp=jtcpdf
[error] Error in dependencies. Installation cancelled.
Installation/upgrade is aborted for the entry point index
Installation ended.

do I have to do it step by step 1.6.9, 1.6.10 etc... ?

Best regards

Philippe

**Comment by laurentj** [comment]
No, for minor version, you don't have to install intermediate release.

Are you using these modules in your application ?

I will investigate on this issue this evening.

**Comment by laurentj** [comment]
Did you take the source code from the download page, or from the repository ? (source from the repository in the 1.6 branch and lower cannot be used directly, there is a "build" to do)

**Comment by migratis** [comment]
Yes I took the source from the download page and I use those plugins jAcl2, jAcl2db and jTcpdf

**Comment by laurentj** [comment]
What are the values of jacl2.version, jacl2db.version and jtcpdf.version in your installer.ini.php? are they empty ? Are there other empty *.version values?

**Comment by migratis** [comment]
I think you've pointed it; all jacl2.version, jacl2db.version and jtcpdf.version are empty, what version do I have to put ? The first install should have filled thoose values isn't it ?

**Comment by laurentj** [comment]
Yes, they should contain the previous version of these modules. 

```
jacl2.version = 1.5
jacl2db.version = 1.5.2
jtcpdf.version = 1.5
```

If you have other empty version, download the jelix 1.6.8 archive and see the version to set in corresponding module.xml. 

I'm trying to understand why these values are empty and improve the installer to deal with them.

**Comment by laurentj** [comment]
Before the upgrade from 1.6.8 to 1.6.16, what is the history of the upgrade of your application? which versions of Jelix did you use with this app before 1.6.8? or did this app always use jelix 1.6.8? If there are previous upgrades to 1.6.8, did you notice some errors during the upgrade?

**Comment by laurentj** [comment]
For the issue about the upgrade it self, the commit I made will improve error messages. Since your installer.ini.php did not contain good values, I don't see a way to fix that. 

For your error about tcpdf : it seems the installer of the jtcpdf module was executed during the installation, and so has changed the entrypoint configuration file with the default value `tcpdf`. Since the entry point config file has priority over the mainconfig file, your setting was not taken account. I fixed the installer of the jtcpdf module  in order to write the default value into mainconfig, and only if a value is not already existing. 

However, your message is strange because it says that the file lib/jelix-modules/jtcpdf/responses/jResponseTcpdf.class.php does not exists, whereas the module jtcpdf is here since the jtcpdf installer wrote somewhere in your config files  `tcpdf = jtcpdf~jResponseTcpdf`... Did you remove the jResponseTcpdf class from `lib/jelix-modules/jtcpdf/responses/` ?
